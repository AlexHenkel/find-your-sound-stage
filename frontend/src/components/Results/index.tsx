import { H2, H3 } from '@blueprintjs/core';
import Container from 'components/Container';
import Result from 'components/Result';
import { StudioResult } from 'utils/api';
import styles from './styles.module.css';

interface Props {
  response?: StudioResult[];
}

export default function Results({ response }: Props) {
  return (
    <Container>
      {response ? (
        response.length ? (
          <>
            <H2>Results</H2>
            <div className={styles.results}>
              {response.map(result => (
                <Result
                  key={`${result.id}-${result.availableStage.stageId}`}
                  {...result}
                ></Result>
              ))}
            </div>
          </>
        ) : (
          <H3>No results ðŸ˜” Please try again with different filters</H3>
        )
      ) : (
        <H3>Fill above filters and start your search ðŸ‘†</H3>
      )}
    </Container>
  );
}
