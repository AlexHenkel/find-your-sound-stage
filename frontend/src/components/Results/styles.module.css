.results {
  display: grid;
  grid-template-columns: 1fr;
  gap: 20px;
}

@media screen and (min-width: 560px) {
  .results {
    grid-template-columns: repeat(auto-fill, minmax(400px, 1fr));
  }
}
