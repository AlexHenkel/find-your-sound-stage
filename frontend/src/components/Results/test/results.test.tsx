import { render } from '@testing-library/react';
import Results from '..';
import { mockResponse } from './mock';

describe('<Results />', () => {
  test('Render correct results', () => {
    const { container } = render(<Results response={mockResponse} />);
    expect(container).toMatchSnapshot();
  });
});
