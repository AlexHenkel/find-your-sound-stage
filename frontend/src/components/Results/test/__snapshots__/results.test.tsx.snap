// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> Render correct results 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="bp3-heading"
    >
      Results
    </h2>
    <div
      class="results"
    >
      <div
        class="bp3-card bp3-interactive bp3-elevation-2"
      >
        <h3
          class="bp3-heading"
        >
          Sunset Bronson
           - 
          1A
        </h3>
        <p
          class="location"
        >
          Los Angeles
          , 
          California
          , 
          United States of America
        </p>
      </div>
      <div
        class="bp3-card bp3-interactive bp3-elevation-2"
      >
        <h3
          class="bp3-heading"
        >
          Screen Gems Studio
           - 
          1
        </h3>
        <p
          class="location"
        >
          Atlanta
          , 
          Georgia
          , 
          United States of America
        </p>
      </div>
      <div
        class="bp3-card bp3-interactive bp3-elevation-2"
      >
        <h3
          class="bp3-heading"
        >
          Screen Gems Studio
           - 
          1A
        </h3>
        <p
          class="location"
        >
          Atlanta
          , 
          Georgia
          , 
          United States of America
        </p>
      </div>
    </div>
  </div>
</div>
`;
